### PR TITLE
Issue 687

### LINKED ISSUE #687: glog is dead, move to klog
The Kuberentes community has moved on from glog to their own implementation cleverly named (wait for it) "klog".

The problem was that glog wasn't actively maintained and had some shortcomings and was difficult to maintain and keep up to date for the K8s project.  So, we have klog which is under active development and is a drop in replacement for glog.  We should update our glog usage to klog so we can keep moving forward and don't get stuck, and because it's always nice to be consistent in the eco-system.

### PR DESCRIPTION
**Pull request by u5surf**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #687 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by awels** [comment]
ci test please

File: glide.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package: kubevirt.io/containerized-data-importer
 import:
-- package: github.com/golang/glog
+- package: k8s.io/klog
```
Comment (awels) [inline glide.yaml]: Don't forget to do
```bash
glide cc && glide up -v
```
That will update all the required dependencies. That will add a ton of files, which is fine.

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
As long as you don't update the dependencies in this patch, the CI will fail due to missing dependencies.
Run glide cc && glide up -v and then add the resulting changes in vendor to this patch.

**Comment by u5surf** [comment]
@awels I tried to run glide cc & glide up -v, but it’s failed several reasons which seem to have nothing to do with this updating,
Although I tried upstream master branch, it failed same reasons.

**Comment by awels** [comment]
@u5surf Did you check out the code in $GOPATH/src/kubevirt.io/containerized-data-importer? If not that is probably the source of the problem. I created and merge a dependency update today so I know it works.

**Comment by u5surf** [comment]
@awels 
Here is a piece of result for `glide cc && glide up -v` in localhost
```
$glide cc && glide up -v
[INFO]  Glide cache has been cleared.
[WARN]  The name listed in the config file (kubevirt.io/containerized-data-importer) does not match the current location (github.com/kubevirt/containerized-data-importer)
[INFO]  Downloading dependencies. Please wait...
[INFO]  --> Fetching k8s.io/klog
[INFO]  --> Fetching github.com/gorilla/mux
[INFO]  --> Fetching k8s.io/apimachinery
[INFO]  --> Fetching k8s.io/kube-aggregator
[INFO]  --> Fetching github.com/emicklei/go-restful
[INFO]  --> Fetching gopkg.in/square/go-jose.v2
[INFO]  --> Fetching github.com/prometheus/client_golang
[INFO]  --> Fetching github.com/emicklei/go-restful-openapi
[INFO]  --> Fetching k8s.io/apiextensions-apiserver
[INFO]  --> Fetching github.com/kelseyhightower/envconfig
[INFO]  --> Fetching sigs.k8s.io/controller-runtime
[INFO]  --> Fetching github.com/openshift/api
[INFO]  --> Fetching k8s.io/client-go
[INFO]  --> Fetching kubevirt.io/qe-tools
[INFO]  --> Fetching github.com/minio/minio-go
[INFO]  --> Fetching github.com/spf13/pflag
[INFO]  --> Fetching github.com/xi2/xz
[INFO]  --> Fetching k8s.io/code-generator
[INFO]  --> Fetching k8s.io/gengo
[INFO]  --> Setting version for github.com/modern-go/reflect2 to 05fbef0ca5da472bbf96c9322b84a53edc03c9fd.
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/core/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/core/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/upload/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/upload/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/core/v1alpha1/fake: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/core/v1alpha1/fake
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/upload/v1alpha1/fake: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/upload/v1alpha1/fake
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/scheme: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/scheme
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/fake: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/fake
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/core: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/core
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/internalinterfaces: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/internalinterfaces
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/upload: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/upload
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/core/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/core/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/upload/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/upload/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/controller: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/controller
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/resources/cluster: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/resources/cluster
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/resources/namespaced: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/resources/namespaced
[INFO]  --> Fetching github.com/mattbaird/jsonpatch
...
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/resources/namespaced: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/resources/namespaced
[INFO]  --> Fetching github.com/mattbaird/jsonpatch
[INFO]  --> Fetching go.uber.org/atomic
[INFO]  --> Detected semantic version. Setting version for go.uber.org/atomic to v1.3.2
[INFO]  --> Fetching go.uber.org/multierr
[INFO]  --> Detected semantic version. Setting version for go.uber.org/multierr to v1.1.0
[INFO]  --> Fetching github.com/go-openapi/jsonpointer
[INFO]  --> Setting version for github.com/go-openapi/jsonpointer to 46af16f9f7b149af66e5d1bd010e3574dc06de98.
[INFO]  --> Fetching github.com/go-openapi/jsonreference
[INFO]  --> Setting version for github.com/go-openapi/jsonreference to 13c6e3589ad90f49bd3e3bbe2c2cb3d7a4142272.
[INFO]  --> Fetching github.com/go-openapi/swag
[INFO]  --> Setting version for github.com/go-openapi/swag to f3f9494671f93fcff853e3c6e9e948b3eb71e590.
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/apis/core: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-apis-core
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/apis/upload/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-apis-upload/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/keys/keystest: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-keys/keystest
[INFO]  --> Fetching github.com/go-ini/ini
[INFO]  --> Fetching github.com/mitchellh/go-homedir
[INFO]  --> Fetching golang.org/x/text
[INFO]  --> Setting version for golang.org/x/text to b19bf474d317b857955b12035d2c5acb57ce8b01.
[INFO]  --> Fetching github.com/docker/spdystream
[INFO]  --> Setting version for github.com/docker/spdystream to 449fdfce4d962303d702fec724ef0ad181c92528.
[INFO]  --> Fetching github.com/modern-go/concurrent
[INFO]  --> Setting version for github.com/modern-go/concurrent to bacd9c7ef1dd9b15be4a9909b8ac7a4e313eec94.
[INFO]  --> Fetching github.com/PuerkitoBio/purell
[INFO]  --> Setting version for github.com/PuerkitoBio/purell to 8a290539e2e8629dbc4e6bad948158f790ec31f4.
[INFO]  --> Fetching github.com/mailru/easyjson
[INFO]  --> Setting version for github.com/mailru/easyjson to 2f5df55504ebc322e4d52d34df6a1f5b503bf26d.
[INFO]  --> Fetching github.com/PuerkitoBio/urlesc
[INFO]  --> Setting version for github.com/PuerkitoBio/urlesc to 5bd2802263f21d8788851d5305584c82a5c75d7e.
[ERROR] Failed to retrieve a list of dependencies: Error resolving imports
```

The master branch was failed explicitly...
```
$git log -n 1
commit 187dc1213299fc3e4dc246e6bd12da95eec19d93 (HEAD -> master, origin/master, origin/HEAD)
Merge: 6b997b5b 318fca8d
Author: Alexander Wels <awels@redhat.com>
Date:   Tue Mar 5 17:02:30 2019 -0500

    Merge pull request #693 from awels/update_deps

    Update dependencies to latest

$glide cc && glide up -v
[INFO]  Glide cache has been cleared.
[WARN]  The name listed in the config file (kubevirt.io/containerized-data-importer) does not match the current location (github.com/kubevirt/containerized-data-importer)
[INFO]  Downloading dependencies. Please wait...
[INFO]  --> Fetching github.com/gorilla/mux
[INFO]  --> Fetching k8s.io/apimachinery
[INFO]  --> Fetching k8s.io/code-generator
[INFO]  --> Fetching github.com/openshift/api
[INFO]  --> Fetching github.com/emicklei/go-restful-openapi
[INFO]  --> Fetching github.com/prometheus/client_golang
[INFO]  --> Fetching sigs.k8s.io/controller-runtime
[INFO]  --> Fetching k8s.io/kube-aggregator
[INFO]  --> Fetching github.com/emicklei/go-restful
[INFO]  --> Fetching k8s.io/apiextensions-apiserver
[INFO]  --> Fetching github.com/kelseyhightower/envconfig
[INFO]  --> Fetching gopkg.in/square/go-jose.v2
[INFO]  --> Fetching github.com/ulikunitz/xz
[INFO]  --> Fetching github.com/golang/glog
[INFO]  --> Fetching k8s.io/client-go
[INFO]  --> Fetching k8s.io/gengo
[INFO]  --> Fetching github.com/spf13/pflag
[INFO]  --> Fetching kubevirt.io/qe-tools
[INFO]  --> Fetching github.com/xi2/xz
[INFO]  --> Fetching github.com/minio/minio-go
[INFO]  --> Fetching github.com/openshift/client-go
[INFO]  --> Setting version for github.com/emicklei/go-restful to v2.6.0.
[INFO]  --> Setting version for github.com/openshift/api to release-3.11.
[INFO]  --> Setting version for kubevirt.io/qe-tools to 13fc8573324cd2af42c88043b5a7c6e03bf39f95.
[INFO]  --> Setting version for github.com/openshift/client-go to release-3.11.
[INFO]  --> Setting version for k8s.io/client-go to kubernetes-1.12.4.
[INFO]  --> Setting version for k8s.io/apimachinery to kubernetes-1.12.4.
[INFO]  --> Detected semantic version. Setting version for github.com/minio/minio-go to 4.0.9
[INFO]  --> Detected semantic version. Setting version for github.com/emicklei/go-restful-openapi to v1.0.0
[INFO]  --> Detected semantic version. Setting version for github.com/gorilla/mux to v1.7.0
[INFO]  --> Detected semantic version. Setting version for github.com/kelseyhightower/envconfig to v1.3.0
[INFO]  --> Detected semantic version. Setting version for github.com/prometheus/client_golang to v0.9.2
[INFO]  --> Setting version for k8s.io/kube-aggregator to kubernetes-1.12.4.
...

[INFO]  --> Setting version for github.com/modern-go/reflect2 to 05fbef0ca5da472bbf96c9322b84a53edc03c9fd.
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/core/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/core/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/upload/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/upload/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/core/v1alpha1/fake: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/core/v1alpha1/fake
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/typed/upload/v1alpha1/fake: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/typed/upload/v1alpha1/fake
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/scheme: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/scheme
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned/fake: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-clientset-versioned/fake
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/core: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/core
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/internalinterfaces: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/internalinterfaces
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/upload: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/upload
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/core/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/core/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions/upload/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-client-informers-externalversions/upload/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/controller: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/controller
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/resources/cluster: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/resources/cluster
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/operator/resources/namespaced: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-operator/resources/namespaced
[INFO]  --> Fetching github.com/mattbaird/jsonpatch
[INFO]  --> Fetching go.uber.org/atomic
[INFO]  --> Detected semantic version. Setting version for go.uber.org/atomic to v1.3.2
[INFO]  --> Fetching go.uber.org/multierr
[INFO]  --> Detected semantic version. Setting version for go.uber.org/multierr to v1.1.0
[INFO]  --> Fetching github.com/go-openapi/jsonpointer
[INFO]  --> Setting version for github.com/go-openapi/jsonpointer to 46af16f9f7b149af66e5d1bd010e3574dc06de98.
[INFO]  --> Fetching github.com/go-openapi/jsonreference
[INFO]  --> Setting version for github.com/go-openapi/jsonreference to 13c6e3589ad90f49bd3e3bbe2c2cb3d7a4142272.
[INFO]  --> Fetching github.com/go-openapi/swag
[INFO]  --> Setting version for github.com/go-openapi/swag to f3f9494671f93fcff853e3c6e9e948b3eb71e590.
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/apis/upload/v1alpha1: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-apis-upload/v1alpha1
[ERROR] Error scanning kubevirt.io/containerized-data-importer/pkg/keys/keystest: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-pkg-keys/keystest
[INFO]  --> Fetching github.com/go-ini/ini
[INFO]  --> Fetching github.com/mitchellh/go-homedir
[ERROR] Error scanning kubevirt.io/containerized-data-importer/tests/utils: cannot find package "." in:
        /Users/yugo-horie/.glide/cache/src/https-kubevirt.io-containerized-data-importer-tests/utils
[INFO]  --> Fetching golang.org/x/text
[INFO]  --> Setting version for golang.org/x/text to b19bf474d317b857955b12035d2c5acb57ce8b01.
[INFO]  --> Fetching github.com/docker/spdystream
[INFO]  --> Setting version for github.com/docker/spdystream to 449fdfce4d962303d702fec724ef0ad181c92528.
[INFO]  --> Fetching github.com/modern-go/concurrent
[INFO]  --> Setting version for github.com/modern-go/concurrent to bacd9c7ef1dd9b15be4a9909b8ac7a4e313eec94.
[INFO]  --> Fetching github.com/PuerkitoBio/purell
[INFO]  --> Setting version for github.com/PuerkitoBio/purell to 8a290539e2e8629dbc4e6bad948158f790ec31f4.
[INFO]  --> Fetching github.com/mailru/easyjson
[INFO]  --> Setting version for github.com/mailru/easyjson to 2f5df55504ebc322e4d52d34df6a1f5b503bf26d.
[INFO]  --> Fetching github.com/PuerkitoBio/urlesc
[INFO]  --> Setting version for github.com/PuerkitoBio/urlesc to 5bd2802263f21d8788851d5305584c82a5c75d7e.
[ERROR] Failed to retrieve a list of dependencies: Error resolving imports
$
```

**Comment by awels** [comment]
@u5surf I think you have the code checked out somewhere that is NOT $GOPATH/src/kubevirt.io/containerized-data-importer I just tried glide cc && glide up -v in a different directory and got the same errors.

**Comment by awels** [comment]
Looks like the linter is unhappy with the new klog flag parsing code. Looks fine to me, but the easiest way to fix it:
gofmt -w  tools/cdi-func-test-file-host-init/main.go  tools/cdi-func-test-registry-init/main.go

again make test-lint will catch these.

**Comment by u5surf** [comment]
ci test please

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
Of course they said klog is an in place replacement of glog, and they lied. Apparently the flag parsing for klog is behind that of glog, so if you have any libraries that use glog, it will eat the flags. I think you noticed this already for some of the containers, since you wrote this:
```go
klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
klog.InitFlags(klogFlags)
flag.CommandLine.VisitAll(func(f1 *flag.Flag) {
	f2 := klogFlags.Lookup(f1.Name)
	if f2 != nil {
		value := f1.Value.String()
		f2.Value.Set(value)
	}
})
```
That copies the flags into the klog flags.

We are going to have to do that for basically all our containers to make the logging work correctly for them. In particular the ones that are failing the tests are cmd/cdi-importer/importer.go and cmd/cdi-controller/controller.go

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
The one failure is flaky test, so I think this looks good. Thanks @u5surf

**Pull request merged**